src/parser/druid/balance/modules/talents/azeritetraits/ArcanicPulsar.tsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import SPELLS from 'common/SPELLS/index';
+
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+
+import Statistic from 'interface/statistics/Statistic';
+import Events, { CastEvent, DamageEvent } from 'parser/core/Events';
+import STATISTIC_ORDER from 'interface/others/STATISTIC_ORDER';
+import BoringSpellValueText
+  from 'interface/statistics/components/BoringSpellValueText';
+import ItemDamageDone from 'interface/ItemDamageDone';
+import { calculateAzeriteEffects } from 'common/stats';
+
+class ArcanicPulsar extends Analyzer {
+  /**
+   * Starsurge's damage is increased by 2181. Every 9 Starsurges, gain
+   * Celestial Alignment for 6 sec.
+   *
+   * Example Log:
+   * https://www.warcraftlogs.com/reports/ZfcqAWCn7gHtp49G#fight=17&type=auras&ability=287790
+   *
+   */
+
+  protected damageGained = 0;
+  protected bonusDamage = 0;
+  protected firstStarsurgeCast = true;
+  protected celestialAlignmentCount = 0;
+
+  constructor(options: any) {
+    super(options);
+    this.active
+      = this.selectedCombatant.hasTrait(SPELLS.ARCANIC_PULSAR_TRAIT.id);
+
+    if (!this.active) {
+      return;
+    }
+
+    this.bonusDamage
+      = this.selectedCombatant.traitsBySpellId[SPELLS.ARCANIC_PULSAR_TRAIT.id]
+      .reduce((total, rank) => {
+        const [damage] = calculateAzeriteEffects(
+          SPELLS.ARCANIC_PULSAR_TRAIT.id,
+          rank,
+        );
+        return total + damage;
+      }, 0);
```

Comment:
You can't calculate gained damage like that. It's much more intricate to calculate the actual gained damage, see my implementation of something similiar here: https://github.com/WoWAnalyzer/WoWAnalyzer/blob/master/src/parser/hunter/beastmastery/modules/spells/azeritetraits/FeedingFrenzy.tsx#L133

Reply:
Interesting aproach, Took me some time to figure out what your 2nd used coefficient was (FEEDING_FRENZY_DAMAGE_COEFFICIENT)  since that was changed since it was changed to 0.216. in v8.2.

Reply:
Your solution is still lacking versatility scaling, tauren scaling and anything else that would affect spell damage (such as boss armor etc)

Reply:
See here for a longer, more commented out version than mine: https://github.com/WoWAnalyzer/WoWAnalyzer/blob/master/src/parser/druid/feral/modules/azeritetraits/WildFleshrending.js

Reply:
Do you mean the version in the latest commits, on the one when you first reviewed that part of code?